src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -119,32 +130,53 @@ class PalantirDockerPlugin implements Plugin<Project> {
                 logging.captureStandardError LogLevel.ERROR
             }
 
-            if (!ext.tags.isEmpty()) {
+            Map<String, Object> tags = ext.tags.collectEntries { taskName, tagName ->
+                [generateTagTaskName(taskName), [
+                        rawTagName          : tagName,
+                        resolveTagNameAction: { -> tagName }
```

Comment:
I would prefer to make this a defined immutable object, but that might have to wait until we convert this to Java

Reply:
I can't make this map as immutable, because unresolved tag will add to this map later.